pkg/controllers/kepler_internal.go

```diff
@@ -74,12 +77,39 @@ func (r *KeplerInternalReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&rbacv1.ClusterRoleBinding{}, genChanged).
 		Owns(&rbacv1.ClusterRole{}, genChanged)
 
+	c = c.Watches(&corev1.Secret{},
+		handler.EnqueueRequestsFromMapFunc(r.mapSecretToRequests))
+
 	if Config.Cluster == k8s.OpenShift {
 		c = c.Owns(&secv1.SecurityContextConstraints{}, genChanged)
 	}
 	return c.Complete(r)
 }
 
+// mapSecretToRequests returns the reconcile requests for kepler-internal objects for which an associated redfish secret has been changed.
+func (r *KeplerInternalReconciler) mapSecretToRequests(ctx context.Context, object client.Object) []reconcile.Request {
+
+	secret, ok := object.(*corev1.Secret)
+	if !ok {
+		return nil
+	}
+
+	ks := v1alpha1.KeplerInternalList{}
+	if err := r.List(ctx, &ks); err != nil {
```

Comment:
I think we can improve the perf a bit more by using the `fieldSelector`
see: https://book.kubebuilder.io/reference/watching-resources/externally-managed